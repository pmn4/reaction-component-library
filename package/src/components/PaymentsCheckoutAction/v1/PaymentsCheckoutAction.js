import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { CustomPropTypes, addTypographyStyles, applyTheme } from "../../../utils";

const SubTitle = styled.div`
  ${addTypographyStyles("PaymentCheckoutAction", "labelTextSemiBold")};
`;

const Title = styled.div`
  ${addTypographyStyles("PaymentCheckoutAction", "subheadingTextSemiBold")};
`;

const billingAddressOptions = [{
  id: "1",
  label: "Same as shipping address",
  value: "same_as_shipping"
},
{
  id: "2",
  label: "Use a different billing address",
  value: "use_different_billing_address"
}];

const options = [{
  id: "5",
  label: "American Express",
  detail: "ending in 4040",
  value: "amex"
},
{
  id: "6",
  label: "Cash",
  value: "cash"
}];

const otherOptions = [{
  id: "7",
  label: "PayPal",
  value: "paypal"
}];

class PaymentsCheckoutAction extends Component {
  static propTypes = {
    /**
     * You can provide a `className` prop that will be applied to the outermost DOM element
     * rendered by this component. We do not recommend using this for styling purposes, but
     * it can be useful as a selector in some situations.
     */
    className: PropTypes.string,
    /**
     * If you've set up a components context using
     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({
      Button: CustomPropTypes.component.isRequired,
      SelectableList: CustomPropTypes.component.isRequired
    }).isRequired
  };

  static defaultProps = {

  };

  render() {
    const { components: { Button, SelectableList } } = this.props;
    const readMore = (
      <Button title="Default" className="myBtn" isTextOnly isShortHeight>Add a new card</Button>
    );
    return (
      <Fragment>
        <SubTitle>Credit cards</SubTitle>
        <SelectableList
          isBordered
          isLeftAligned
          options={options}
          name="PaymentForm"
          listAction={readMore}
        />
        <SubTitle>Other payment methods</SubTitle>
        <SelectableList
          isBordered
          isLeftAligned
          options={otherOptions}
          name="PaymentForm"
        />
        <Title>Billing address</Title>
        <SelectableList
          options={billingAddressOptions}
          name="billingAddressForm"
          value={""}
        />
      </Fragment>
    );
  }
}

export default withComponents(PaymentsCheckoutAction);
