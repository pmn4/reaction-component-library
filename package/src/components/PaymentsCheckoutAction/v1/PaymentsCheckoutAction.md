### Overview

### Usage

```jsx
<PaymentsCheckoutAction />
```

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop | Default | Description |
