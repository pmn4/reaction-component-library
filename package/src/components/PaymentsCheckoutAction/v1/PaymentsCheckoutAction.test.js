import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PaymentsCheckoutAction from "./PaymentsCheckoutAction";

test("basic snapshot", () => {
  const component = renderer.create(<PaymentsCheckoutAction />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
