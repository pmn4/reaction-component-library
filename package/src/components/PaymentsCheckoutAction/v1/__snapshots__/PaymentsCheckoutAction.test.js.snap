// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
Array [
  .c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  line-height: 24px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
}

<div
    className="c0"
  >
    Credit cards
  </div>,
  .c0 {
  margin-top: 5px;
  margin-bottom: 15px;
}

<div
    className="c0"
  >
    SelectableList(undefined)
  </div>,
  .c0 {
  -webkit-font-smoothing: antialiased;
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .02em;
  -moz-letter-spacing: .02em;
  -ms-letter-spacing: .02em;
  letter-spacing: .02em;
  line-height: 1.25;
  line-height: 24px;
  -webkit-letter-spacing: 0.3px;
  -moz-letter-spacing: 0.3px;
  -ms-letter-spacing: 0.3px;
  letter-spacing: 0.3px;
}

<div
    className="c0"
  >
    Other payment methods
  </div>,
  .c0 {
  margin-top: 5px;
  margin-bottom: 15px;
}

<div
    className="c0"
  >
    SelectableList(undefined)
  </div>,
  .c0 {
  -webkit-font-smoothing: antialiased;
  color: #505558;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 18px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 600;
  -webkit-letter-spacing: .03em;
  -moz-letter-spacing: .03em;
  -ms-letter-spacing: .03em;
  letter-spacing: .03em;
  line-height: 1.25;
  padding-bottom: 10px;
  padding-top: 15px;
}

<div
    className="c0"
  >
    Billing address
  </div>,
  <div
    className=""
  >
    SelectableList(undefined)
  </div>,
]
`;
